# Ejercicios de probabilidad

## Ejercicio 1 

Dos dados se lanzan una vez y se observa el total obtenido. Usa una simulación para encontrar la probabilidad estimada de que la puntuación sea mayor a 7 o un número par. Una simulación es una repetición del mismo experimento multiples veces para observar su comportamiento:

- Ejecuta el experimento 1000 veces (lanza 2 dados 1000 veces, y suma el número de ambos dados).
- Lleva ua cuenta de los números y las veces que la suma fue mayor a 7 o un número par.
- Divide el número del paso 2 entre el número de iteraciones(1000).

In [7]:
import random

In [9]:
# Número de experimentos (lanzamientos)
num_experiments = 1000

# Contador para las veces que se cumple la condición
count = 0

# Realizamos la simulación
for _ in range(num_experiments):
    # Simular el lanzamiento de dos dados (valores entre 1 y 6)
    dice_1 = random.randint(1, 6)
    dice_2 = random.randint(1, 6)
    
    # Sumar los dos valores de los dados
    total = dice_1 + dice_2
    
    # Verificar si la suma es mayor a 7 o es un número par
    if total > 7 or total % 2 == 0:
        count += 1

# Calcular la probabilidad
probabilidad = count / num_experiments

# Mostrar el resultado
round(probabilidad, 2)



0.63

# Conclusion:
> La probabilidad de que la suma sea mayor a 7 o par es 0,69

## Ejercicio 2

Una caja contiene 10 bolas blancas, 20 bolas rojas y 30 bolas verdes. Si tomamos 5 bolas de la caja reemplazándolas (tomando una bola, anotando el color y luego regresandola a la caja). Queremos saber la probabilidad de:

1. Tomar 3 blancas y 2 rojas.
2. Toas todas del mismo color.

Ejecuta el experimento 1000 veces y calcula las probabilidades que mencionadas.

In [10]:
import random

In [14]:
ball_box = {}

# Crea la cajas con las bolas
for i in range(60):
    if i < 10:
        ball_box[i] = "White"
    elif (i > 9) and (i < 30):
        ball_box[i] = "Red"
    else:
        ball_box[i] = "Green"


# Número de experimentos
num_experiments = 1000

# Contadores para las probabilidades
count_3_blancas_2_rojas = 0
count_todas_mismo_color = 0

# Realizamos la simulación
for _ in range(num_experiments):
    # Extraemos 5 bolas con reemplazo (usamos random.choice() para elegir índices aleatorios)
    extraccion = [ball_box[random.choice(list(ball_box.keys()))] for _ in range(5)]
    
    # Contamos cuántas bolas blancas y rojas hay en la extracción
    blancas = extraccion.count('White')
    rojas = extraccion.count('Red')
    
    # Verificamos si hay 3 bolas blancas y 2 bolas rojas
    if blancas == 3 and rojas == 2:
        count_3_blancas_2_rojas += 1
    
    # Verificamos si todas las bolas son del mismo color
    if len(set(extraccion)) == 1:
        count_todas_mismo_color += 1

# Calculamos las probabilidades
prob_3_blancas_2_rojas = count_3_blancas_2_rojas / num_experiments
prob_todas_mismo_color = count_todas_mismo_color / num_experiments

# Imprimimos los resultados
round(prob_3_blancas_2_rojas, 3), round(prob_todas_mismo_color, 3)



(0.006, 0.043)

# Conclusion:
>La probabilidad de tomar 3 blancas y 2 rojas: 0.006. La probabilidad de tomar todas del mismo color: 0.043